src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -366,6 +346,94 @@ private ConnectionState doReadHTTPInitial(HttpRequest httpRequest) {
         }
     }
 
+    @Sharable
+    protected class ClientToProxyMessageProcessor extends ChannelInboundHandlerAdapter {
```

Comment:
this handler contains authentication and payload transformation logic which is executed in a separate executor service

Reply:
could we move it to a separate class / java file?

Reply:
this class encapsulate some code of client to proxy connection.. it is easier to have it here..